### PR TITLE
feat: celestia-app docker document

### PR DESCRIPTION
**Pull request by ogunleye0720**
## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

## Summary by CodeRabbit

- **Documentation**
	- Expanded `docker-images.md` with a comprehensive guide for setting up Celestia-App and Celestia-Node using Docker, featuring improved navigation, detailed prerequisites, step-by-step instructions, and additional visual resources for user assistance.

### CODE REVIEW

**Review by jcstein — CHANGES_REQUESTED** [review]
needs to be formatted properly and linted

File: nodes/celestia-app-docker-image.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+   ```bash [linux or unix OS]
+   cd $HOME
+   ```
+2. Create a file in which the keyring would be stored. The file will be mounted as a volume into the docker container.
+   ```bash [linux or unix OS]
+   Home/ touch .celestia-app
+   ```
+3. Using a suitable text editor of your choice, open the .celestial-app file and paste the keyring of the prefunded account.
+   
+5. We recommend that you set the necessary file permission for the .celestial-app file. A simple read access is all that is required for the docker container to access the content of the file.
```
Comment (jcstein) [inline nodes/celestia-app-docker-image.md:31]: 4?

File: nodes/celestia-app-docker-image.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+   
+7. You can run the txsim Docker image using the docker run command below.
```
Comment (jcstein) [inline nodes/celestia-app-docker-image.md:33]: 5?

File: nodes/celestia-app-docker-image.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+   ```bash [linux or unix OS}
+   docker run -it -v $HOME/.celestia-app:/home/celestia ghcr.io/celestiaorg/txsim -k 0 -r http://consensus-validator-robusta-rc6.celestia-robusta.com:26657,http://consensus-full-robusta-rc6.celestia-robusta.com:26657 -g consensus-validator-robusta-rc6.celestia-robusta.com:9090 -t 10s -b 10 -d 100 -e 10
+   ```
+8. In this command, the -v option is used to mount the $HOME/.celestia-app directory from the host to the /home/celestia directory in the Docker container. This allows the txsim binary to access the keyring for the prefunded account.
+
+### Congratulations !!! You have successfuly set up the celestia app 😎
```
Comment (jcstein) [inline nodes/celestia-app-docker-image.md:39]:
```suggestion
Congratulations! You have successfuly set up celestia-app in Docker 😎
```

Comment (ogunleye0720) [reply nodes/celestia-app-docker-image.md:39]: Hello @jcstein, Thank you very much for the feedback. I'll make the necessary adjustments immediately.

**Comment by jcstein** [comment]
what OS have you tested this on @ogunleye0720 ?

**Comment by ogunleye0720** [comment]
> what OS have you tested this on @ogunleye0720 ?


Linux

**Comment by jcstein** [comment]
[] this pr needs to pass linting and formatting rules with `yarn lint` and `yarn format`

**Comment by jcstein** [comment]
no worries! thank you. note for myself too :)

i'll also be testing this on mac for final review

**Comment by jcstein** [comment]
gm @ogunleye0720 I apologize for the delay here. can you please work this into the existing docker page, into a section for celestia-app? instead of a new docker page? thanks!

**Comment by jcstein** [comment]
checking in here @ogunleye0720

**Comment by ogunleye0720** [comment]
> checking in here @ogunleye0720

Hello Justein, sorry for the late response. We are experiencing an internet outage here in Africa caused by a subsea cable break. I would start working on this today.

**Comment by jcstein** [comment]
let's combine this with the existing docker page https://docs.celestia.org/nodes/docker-images

**Comment by ogunleye0720** [comment]
> let's combine this with the existing docker page https://docs.celestia.org/nodes/docker-images

Apologies, I thought i was on the other repository. I have issues setting up Markdownlint on the  https://docs.celestia.org/nodes/docker-images repository. Can you help me with any resource on how to set that up ?

**Comment by jcstein** [comment]
this is all in the same repository. i'm just suggesting that you move the contents of the new page that you have to the existing page https://docs.celestia.org/nodes/docker-images which can be found in this repo under /nodes/docker-images.md

**Comment by jcstein** [comment]
don't worry about markdown lint for now, but you can set it up with https://github.com/markdownlint/markdownlint and be sure to just lint the docs themselves, else you will see a lot of linting errors in node_modules

**Comment by ogunleye0720** [comment]
> don't worry about markdown lint for now, but you can set it up with https://github.com/markdownlint/markdownlint and be sure to just lint the docs themselves, else you will see a lot of linting errors in node_modules

So this is what I have done so far, after some reviews, I was asked to some adjustments and make sure it's properly linted. https://github.com/ogunleye0720/celestia-app/tree/celestia-app-docker/.github

**Comment by jcstein** [comment]
awesome! thank you for the context. let's please now add your new additions the existing docker page, so that both celestia-app and celestia-node for docker live in the same page
https://docs.celestia.org/nodes/docker-images

**Comment by jcstein** [comment]
let's remove markdownlint as a dependency though on this!

File: package.json [hunk 4]
```diff
@@ -19,6 +19,7 @@
     "husky": "^8.0.3",
     "i": "^0.3.7",
     "lint-staged": "^14.0.1",
+    "markdownlint": "^0.34.0",
```
Comment (jcstein) [inline package.json:22]: let's remove markdownlint as a dependency though on this and remove package-lock.json as this repo uses yarn not npm

Comment (ogunleye0720) [reply package.json:22]: Do you mi d if we schedule a Google Meet ? This should help me understand better.

Comment (jcstein) [reply package.json:22]: I'll fix this up. All that is left is to please combine the new page you made with the existing docker page.

Comment (ogunleye0720) [reply package.json]: > I'll fix this up. All that is left is to please combine the new page you made with the existing docker page.

All I am trying to understand is which existing page? Is the page on github? If yes post link then I can create a pull request. The link you dropped earlier directed me to a page outside github.

**Comment by jcstein** [comment]
TODO
- [ ] combine the new page in this PR with the existing docker page, so that there is only one page for docker for both celestia-node and celestia-app at `docker-images.md`

File: nodes/celestia-app-docker-image.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
+d
```
Comment (jcstein) [inline nodes/celestia-app-docker-image.md:1]:
```suggestion
```

**Comment by ogunleye0720** [comment]
> TODO
> 
> * [ ]  combine the new page in this PR with the existing docker page, so that there is only one page for docker for both celestia-node and celestia-app at `docker-images.md`

On it 🫡

**Comment by jcstein** [comment]
Do we need to remove `nodes/celestia-app-docker-image.md`?

File: nodes/docker-images.md [hunk 6]
```diff
@@ -1,9 +1,26 @@
 ---
-sidebar_label: Docker images
-description: Running Celestia Node using Docker.
+Sidebar_Label: Docker images
```
Comment (jcstein) [inline nodes/docker-images.md:2]:
```suggestion
sidebar_Label: Docker images
```

File: nodes/docker-images.md [hunk 7]
```diff
@@ -1,9 +1,26 @@
...
+Description: Comprehensive guide for setting up Celestia-App and Celestia-Node using Docker.
```
Comment (jcstein) [inline nodes/docker-images.md:3]:
```suggestion
description: Comprehensive guide for setting up Celestia-App and Celestia-Node using Docker.
```

File: nodes/docker-images.md [hunk 8]
```diff
@@ -1,9 +1,26 @@
 ---
 sidebar_label: Docker images
-description: Running Celestia Node using Docker.
+description: Comprehensive guide for setting up Celestia-App and Celestia-Node using Docker.
 ---
 
-# 🐳 Docker setup
+# Celestia-App and Celestia-Node Setting Up with Docker 🐳
```
Comment (jcstein) [inline nodes/docker-images.md:6]:
```suggestion
# Celestia-app and celestia-node setup with Docker 🐳
```

File: nodes/docker-images.md [hunk 9]
```diff
@@ -1,9 +1,26 @@
 ---
 sidebar_label: Docker images
-description: Running Celestia Node using Docker.
+description: Comprehensive guide for setting up Celestia-App and Celestia-Node using Docker.
```
Comment (jcstein) [inline nodes/docker-images.md:3]:
```suggestion
description: Comprehensive guide for setting up celestia-app and celestia-node using Docker.
```

**Comment by jcstein** [comment]
gm @ogunleye0720 do you want to clean this up and get it merged?

File: nodes/docker-images.md [hunk 10]
```diff
@@ -14,34 +31,125 @@ import mochaVersions from '/.vitepress/constants/mocha_versions.js'
...
 
-This page has instructions to run celestia-node using Docker. If you are
-looking for instructions to run celestia-node using a binary, please
-refer to the [celestia-node page](./celestia-node.md).
+## Introduction
 
-Using Docker is the easiest way to run celestia-node for most
-users. Docker is a containerization platform that allows you to run celestia-node
-in an isolated environment.
+This page provides instructions on how to set up both the Celestia-App and
```
Comment (jcstein) [inline nodes/docker-images.md:36]: - [ ] use celestia-app and celestia-node throughout this document

Comment (jcstein) [inline nodes/docker-images.md:36]:
```suggestion
This page provides instructions on how to set up both the celestia-app and
```

File: nodes/docker-images.md [hunk 11]
```diff
@@ -14,34 +31,125 @@ import mochaVersions from '/.vitepress/constants/mocha_versions.js'
...
+the Celestia-Node using Docker.
```
Comment (jcstein) [inline nodes/docker-images.md:37]:
```suggestion
the celestia-node using Docker.
```

File: nodes/docker-images.md [hunk 12]
```diff
@@ -14,34 +31,125 @@ import mochaVersions from '/.vitepress/constants/mocha_versions.js'
...
+see [celestia-node page](./celestia-node.md) for additional instructions
```
Comment (jcstein) [inline nodes/docker-images.md:38]: **no it is on this page, is this AI-generated??**

File: nodes/docker-images.md [hunk 13]
```diff
@@ -14,34 +31,125 @@
...
+to run celestia-node using Docker.
```
[the remark this answers is not in the document]
Comment (jcstein) [reply nodes/docker-images.md:39]: sure but this is incorrect anyways

File: nodes/docker-images.md [hunk 14]
```diff
@@ -14,34 +31,125 @@ import mochaVersions from '/.vitepress/constants/mocha_versions.js'
...
 
-This means that you can run celestia-node on your machine without having
-to worry about installing and configuring all of the dependencies required
-to run the node.
+Docker is a container technology that offers a seamless method
+for running applications in isolated environments on any
+operating system.
+
+This means that you can run Celestia-app & celestia-node
```
Comment (jcstein) [inline nodes/docker-images.md:45]:
```suggestion
This means that you can run celestia-app & celestia-node
```

File: nodes/docker-images.md [hunk 15]
```diff
@@ -14,34 +31,125 @@
...
+## Docker Installation
+
+To install Docker on your machine, see the links below for instructions
+on your specific OS.
+
+- [Docker Desktop for Mac or Windows](https://docs.docker.com/get-docker) or
+
+- [Docker Engine for Linux](https://docs.docker.com/engine/install/)
+  
+- Ensure you have a basic understanding of Docker to proceed.
```
[the remark this answers is not in the document]
Comment (jcstein) [reply nodes/docker-images.md:62]: sure

File: nodes/docker-images.md [hunk 16]
```diff
@@ -14,34 +31,125 @@ import mochaVersions from '/.vitepress/constants/mocha_versions.js'
...
+
+## Celestia-App Setup with Docker
```
Comment (jcstein) [inline nodes/docker-images.md:64]: use Celestia-app in this heading

File: nodes/docker-images.md [hunk 17]
```diff
@@ -14,34 +31,125 @@ import mochaVersions from '/.vitepress/constants/mocha_versions.js'
...
+
+This guide provides instructions on how to use the Celestia `txsim` Docker image.
+
+### Overview of Celestia-App `txsim`
```
Comment (jcstein) [inline nodes/docker-images.md:68]: what is this? ive never heard of it? is this ai-generated?

File: nodes/docker-images.md [hunk 18]
```diff
@@ -14,34 +31,125 @@ import mochaVersions from '/.vitepress/constants/mocha_versions.js'
...
+
+The Celestia-App `txsim` binary is a tool that can be
+used to simulate transactions on the Celestia network.
+It can be used to test the performance of the Celestia network.
+The txsim Docker image runs the `txsim` binary with multiple configurable options.
```
Comment (jcstein) [inline nodes/docker-images.md:73]: ????????

File: nodes/docker-images.md [hunk 19]
```diff
@@ -14,34 +31,125 @@ import mochaVersions from '/.vitepress/constants/mocha_versions.js'
...
+2. Create a file in which the keyring would be stored.
+The file would be mounted as a volume into the docker container.
+
+   ```bash
+   touch $HOME/.celestia-app
+   ```
 
-## Quick start
+3. Using a suitable text editor of your choice, open the
+.celestia-app file and paste the keyring of the prefunded account.
```
Comment (jcstein) [inline nodes/docker-images.md:99]: ????

File: nodes/docker-images.md [hunk 20]
```diff
@@ -14,34 +31,125 @@ import mochaVersions from '/.vitepress/constants/mocha_versions.js'
...
+docker container to access the content of the file.
+
+5. You can run the txsim Docker image using the docker run command below.
+
+   ```bash
+   docker run -it \
+   -v $HOME/.celestia-app:/home/celestia ghcr.io/celestiaorg/txsim \
+   -k 0 \
+   -g consensus-validator-robusta-rc6.celestia-robusta.com: \
+   -t 10s -b 10 -d 100 -e 10
```
Comment (jcstein) [inline nodes/docker-images.md:112]:
```suggestion
-B -S
```

File: nodes/docker-images.md [hunk 21]
```diff
@@ -14,34 +31,125 @@ import mochaVersions from '/.vitepress/constants/mocha_versions.js'
...
+Congratulations! You have successfully set up Celestia-App in Docker 😎
+
+### Flag Breakdown
+
+The table below provides a brief explanation of the
+flags used in the docker run command in step 5 of the quick start instructions.
+
+| FLAG | DESCRIPTION | DEFAULT | OPTION |
+| ---- | ---- | ---- | :----: |
+|`-k`|Whether a new key should be created|0|1 for yes, 0 for no|
```
Comment (jcstein) [inline nodes/docker-images.md:129]: going to close this PR. this is all nonsense

File: nodes/docker-images.md [hunk 22]
```diff
@@ -112,15 +220,15 @@ Ubuntu. You can
 
    :::
 
-Congratulations! You now have a celestia-node running!
+Congratulations! You now have a Celestia-Node running 😎
```
Comment (jcstein) [inline nodes/docker-images.md:223]: but i appreciate the emoji 🫡

**Review by jcstein — COMMENTED** [review]
ai-generated mess. closing this PR

**Pull request closed without merging**